packages/lesswrong/lib/collections/users/fragments.ts

```diff
@@ -305,6 +305,47 @@ registerFragment(`
         }
         eaAddedReacts
       }
+      todaysKarmaChanges {
```

Comment:
Sorry I hate doing this, is there a better way to avoid all this duplication?

Reply:
You can do a regular fragment:
```typescript
registerFragment(`
  fragment KarmaChangesBasicInfo on KarmaChangesSimple {
    posts {
      _id
      scoreChange
      title
      slug
      addedReacts {
        reactionType
        userId
      }
      eaAddedReacts
    }
    comments {
      _id
      scoreChange
      description
      postId
      postTitle
      postSlug
      tagSlug
      tagName
      tagCommentType
      addedReacts {
        reactionType
        userId
      }
      eaAddedReacts
    }
    tagRevisions {
      _id
      scoreChange
      tagId
      tagSlug
      tagName
      addedReacts {
        reactionType
        userId
      }
      eaAddedReacts
    }
  }
`)

registerFragment(`
  fragment UserKarmaChanges on User {
    _id
    karmaChanges {
      totalChange
      updateFrequency
      startDate
      endDate
      nextBatchDate
      ...KarmaChangesBasicInfo
      todaysKarmaChanges {
        ...KarmaChangesBasicInfo
      }
    }
  }
`);
```

Reply:
Oh wait actually that results in an error due to a type mismatch. I think you could fix it though by merging these two types:
```
  type KarmaChangesSimple {
    posts: [PostKarmaChange]
    comments: [CommentKarmaChange]
    tagRevisions: [RevisionsKarmaChange]
  }
  type KarmaChanges {
    totalChange: Int
    startDate: Date
    endDate: Date
    nextBatchDate: Date
    updateFrequency: String
    posts: [PostKarmaChange]
    comments: [CommentKarmaChange]
    tagRevisions: [RevisionsKarmaChange]
    todaysKarmaChanges: KarmaChangesSimple
  }
```

Reply:
I tried fixing this via using an interface and that didn't work, so I'm going to give up on this for now :(